_minutes/2023-09-11-2023-09-14-tpac-extra.md

```diff
@@ -0,0 +1,520 @@
+     * [devlin] Yes. Not a handle though.
+     * [oliver] You could use document.querySelector.
+     * [devlin] Yes.
+     * [rob] You could pass a reference to the element via the args parameter.
+     * [devlin] Yuck.
+     * [rob] What's your concern with passing a reference to an element?
+     * [devlin] Complexity of mapping the JS variable to the DOM element in the isolated world, and then back to a JS handle in the main world.
+     * [rob] That should be safe.
+     * [devlin] yes, but there are other problems. … If in my JS I set data on an element, pass a reference to another world, and try to access that data in my other world scfript, would not see the data I set
+     * [rob] That is expected, a normal part of the isolated world concept.
+     * [devlin] Yep, exactly – but it's more surprising if I pass an element reference and get an element reference that those are totally different.  If I pass an ID and then re-fetch that element in the main world, it's more intuitive that they are different values.
+     * [rob] So you want to use structured cloning? Generally a copy of everything? I can agree with that reasoning.
+   * [rob/devlin] DNR: future vision (where do we want it in 5 years)
+     * [oliver] Are we concerned about malicious extensions adding dynamic rules?
+     * [rob] Unless there's something sensitive, let's save this topic for the public meeting.
+     * [devlin] +1
+   * [kiara] browser namespace
+     * [Issue 113](https://github.com/w3c/webextensions/issues/113): Proposal: Make extension APIs browser neutral
+     * [kiara] In short, I asked about supporting the browser global
+     * [oliver] I effectively said we're good with this except for the need to work out what we do with externally connectable
+     * [devlin] Historically, Chrome has done a lot of work to make sure that everything exposed on the global window object be standardized except for the "chrome" namespace, and even then we try to keep it as tight as possible.
+     * [oliver] Can we just keep it as chrome for this specific case?
+     * [tomislav] Talking about two different things, the namespace in extensions and the namespace in web pages.
+     * [devlin] Let's start at the beginning.
+     * [kiara] Can we support "browser" in extension contexts.
+     * [devlin] Potentially, but we would still have the same issue
+     * [simeon] Sounds like we could support chrome and browser. Breaking for developers we can address in manifest version bumps.
+     * [devlin] That introduces a new global, which violates the existing Chrome global variable practices.
+     * [simeon] Even if only in extension contexts?
+     * [devlin] Yes.
+     * [tomislav] lets separate externally_connectable from this, and only focus on the extension world. Safari does not support externally connected right?
+     * [kiara] no, we do support it
+     * [tomislav] oh, ok. I was out of date. Can we discuss the two concepts separately
```

Comment:
We do support it, and had to only expose `window.browser` to pages since in our very limited testing we saw pages thinking Safari was Chrome because `window.chrome` existed.